subprojects/core-api/src/main/java/org/gradle/process/AllJvmArgsAdapter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process;
+
+import org.apache.commons.lang.StringUtils;
+import org.gradle.internal.instrumentation.api.annotations.BytecodeUpgrade;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import static org.gradle.process.internal.AllJvmArgsAdapterUtil.checkDebugConfiguration;
+
+class AllJvmArgsAdapter {
```

Comment:
🤔 Some Javadoc would a short summary of the goal with this adapter would be nice.

Also, shall we call this `JavaForkOptions.AllJvmArgsAdapter`?

Reply:
Moved to `JavaForkOptionsAdapters.AllJvmArgsAdapter `

The reason why this is not in the `JavaForkOptions` is that `JavaForkOptions` in an interface, and thus we can't have package private class there, that is a current requirement for `Adapter` classes